### PR TITLE
Give a warning when the [Benchmark] method is static

### PR DESCRIPTION
**Pull request by Rizzen**
For https://github.com/dotnet/BenchmarkDotNet/issues/983

### CODE REVIEW

**Comment by Rizzen** [comment]
@adamsitnik @AndreyAkinshin We have some static benchmark method cases in InProcessBenchmarkAllCasesReflectionEmitSupported and InProcessBenchmarkAllCasesDelegateCombineSupported. Does it make any sense to have such test cases now?

**Comment by AndreyAkinshin** [comment]
@Rizzen, I think we should fix or remove these tests.

File: tests/BenchmarkDotNet.Tests/Validators/CompilationValidatorTests.cs [hunk 1]
```diff
@@ -28,12 +29,25 @@ public void BenchmarkedMethodNameMustNotContainWhitespaces()
...
+        [InlineData(typeof(BenchmarkClassWithStaticMethod), true)]
+        [InlineData(typeof(BenchmarkClass<PublicClass>), false)]
+        public void Benchmark_Class_Methods_Must_Be_Non_Static(Type type, bool hasErrors)
+        {
+            // Act
+            var validationErrors = CompilationValidator.Default.Validate(BenchmarkConverter.TypeToBenchmarks(type))
+                                                               .ToList();
+            
+            // Assert
+            Assert.Equal(validationErrors.Any(), hasErrors);
```
Comment (AndreyAkinshin) [inline tests/BenchmarkDotNet.Tests/Validators/CompilationValidatorTests.cs]: Please swap the arguments.
The first argument is the expected value; the second argument is the actual value.

**Comment by AndreyAkinshin** [comment]
@Rizzen thanks!

**Pull request merged**